### ISSUE TITLE
ImageAugmentation using tf.keras.preprocessing.image.ImageDataGenerator and tf.datasets: model.fit() is running infinitely

### ISSUE THREAD

**Issue by Thomasaws1** [tensorflow/datasets#2000]
**What I need help with / What I was wondering**
I am facing issue while running the fit() function in TensorFlow(v 2.2.0-rc4) with augmented images(using ImageDataGenerator) passed as a dataset. The fit() function is running infinitely without stopping.

**What I've tried so far**
I tried it with the default code which was shared in Tensorflow documentation. 

Please find the code snippet below:

import tensorflow as tf
from tensorflow.keras.preprocessing.image import ImageDataGenerator
from tensorflow.keras.models import Sequential, Model 
from tensorflow.keras.layers import Dense, Dropout, Flatten
from tensorflow.keras.layers import Conv2D, MaxPooling2D 
from tensorflow.keras.layers import Input, Dense

flowers = tf.keras.utils.get_file(
    'flower_photos',
    'https://storage.googleapis.com/download.tensorflow.org/example_images/flower_photos.tgz',
    untar=True)

img_gen = tf.keras.preprocessing.image.ImageDataGenerator(rescale=1./255, rotation_range=20)

images, labels = next(img_gen.flow_from_directory(flowers))

print(images.dtype, images.shape)
print(labels.dtype, labels.shape)

train_data_gen = img_gen.flow_from_directory(
                    batch_size=32, 
                    directory=flowers,
                    shuffle=True,
                    target_size=(256, 256),
                    class_mode='categorical')

ds = tf.data.Dataset.from_generator(lambda: train_data_gen,
                     output_types=(tf.float32, tf.float32),
                     output_shapes=([32, 256, 256, 3],
                                    [32, 5])
                     )

ds = ds.prefetch(buffer_size=tf.data.experimental.AUTOTUNE)

it = iter(ds)
batch = next(it)
print(batch)

def create_model():
  model = Sequential()
  model.add(Conv2D(32, (3, 3), activation='relu', input_shape=images[0].shape))
  model.add(Conv2D(32, (3, 3), activation='relu'))
  model.add(MaxPooling2D(pool_size=(2, 2)))
  model.add(Dropout(0.5))
  model.add(Conv2D(64, (3, 3), activation='relu'))
  model.add(Conv2D(64, (3, 3), activation='relu'))
  model.add(MaxPooling2D(pool_size=(2, 2)))
  model.add(Dropout(0.5))
  model.add(Flatten())
  model.add(Dense(64, activation='relu'))
  model.add(Dropout(0.5))
  model.add(Dense(5, activation='softmax'))
  return model

model = create_model()
model.compile(loss='categorical_crossentropy', optimizer='rmsprop', metrics=["accuracy"])
model.fit(ds, verbose=1,  batch_size= 32, epochs =1)

This last line of code - fit() is running infinitly without stopping. I had also tried passing steps_per_epoch = total_no_of_train_records/batch_size.

**It would be nice if...**
I would like you to confirm whethere this is a bug in the tensorflow datasets package and in which release will this be fixed.

**Environment information**
* System: Google colaborator
* Python version: v3.6.9
* `tensorflow-datasets`/`tfds-nightly` version: v2.2.0-rc4

**Comment by Conchylicultor** [comment]
Thanks for reporting, however, it seems you're confusing `TFDS` (this library) with the `tf.keras` data generators. Those are two independent projects.

You are not using `import tensorflow_datasets as tfds`, so there is nothing we can do here. You should post your issue in the `tensorflow/tensorflow` repo.

Or if you want to use TFDS, I think you're trying to use this dataset https://www.tensorflow.org/datasets/catalog/tf_flowers

```py
import tensorflow_datasets as tfds

ds = tfds.load('tf_flower', split='train', as_supervised=True)

for img, label in ds:
  print(img.shape, label)
```
